AsyncDisplayKit/ASVideoPlayerNode.mm

```diff
@@ -0,0 +1,729 @@
+{
+  NSDictionary *options;
+  if (_delegateFlags.delegateTimeLabelAttributes) {
+    options = [_delegate videoPlayerNodeTimeLabelAttributes:self timeLabelType:controlType];
+  } else {
+    options = @{
+                NSFontAttributeName : [UIFont systemFontOfSize:12.0],
+                NSForegroundColorAttributeName: _defaultControlsColor
+                };
+  }
+
+
+  NSAttributedString *attributedString = [[NSAttributedString alloc] initWithString:string attributes:options];
+
+  return attributedString;
+}
+
+#pragma mark - ASVideoNodeDelegate
+- (void)videoNode:(ASVideoNode *)videoNode willChangePlayerState:(ASVideoNodePlayerState)state toState:(ASVideoNodePlayerState)toState
+{
+  if (_delegateFlags.delegateVideoNodeWillChangeState) {
+    [_delegate videoPlayerNode:self willChangeVideoNodeState:state toVideoNodeState:toState];
+  }
+
+  if (toState == ASVideoNodePlayerStateReadyToPlay && _durationTextNode) {
+    _duration = _videoNode.currentItem.duration;
+    [self updateDurationTimeLabel];
+  }
+
+  if (toState == ASVideoNodePlayerStatePlaying) {
+    _playbackButtonNode.buttonType = ASDefaultPlaybackButtonTypePause;
+    [self removeSpinner];
+  } else if (toState != ASVideoNodePlayerStatePlaybackLikelyToKeepUpButNotPlaying && toState != ASVideoNodePlayerStateReadyToPlay) {
+    _playbackButtonNode.buttonType = ASDefaultPlaybackButtonTypePlay;
+  }
+
+  if (toState == ASVideoNodePlayerStateLoading || toState == ASVideoNodePlayerStateInitialLoading) {
+    [self showSpinner];
+  }
+
+  if (toState == ASVideoNodePlayerStateReadyToPlay || toState == ASVideoNodePlayerStatePaused || toState == ASVideoNodePlayerStatePlaybackLikelyToKeepUpButNotPlaying) {
+    [self removeSpinner];
+  }
+}
+
+- (BOOL)videoNode:(ASVideoNode *)videoNode shouldChangePlayerStateTo:(ASVideoNodePlayerState)state
+{
+  if (_delegateFlags.delegateVideoNodeShouldChangeState) {
+    return [_delegate videoPlayerNode:self shouldChangeVideoNodeStateTo:state];
+  }
+  return YES;
+}
+
+- (void)videoNode:(ASVideoNode *)videoNode didPlayToTimeInterval:(NSTimeInterval)timeInterval
+{
+  //TODO: ask Max about CMTime problem in ASVideoNode Header file
+  //as we said yesterday, we must use CMTime in ASVideoNode instead of NSTimeInterval
+  //when this will be done, must just proxy value to delegate
+  if (_delegateFlags.delegateVideoNodeDidPlayToTime) {
+    [_delegate videoPlayerNode:self didPlayToTime:_videoNode.player.currentTime];
+  }
+
+  if (_isSeeking) {
+    return;
+  }
+
+  if (_elapsedTextNode) {
+    [self updateElapsedTimeLabel:timeInterval];
+  }
+
+  if (_scrubberNode) {
+    [(UISlider*)_scrubberNode.view setValue:( timeInterval / CMTimeGetSeconds(_duration) ) animated:NO];
+  }
+}
+
+- (void)videoDidPlayToEnd:(ASVideoNode *)videoNode
+{
+  if (_delegateFlags.delegateVideoNodePlaybackDidFinish) {
+    [_delegate videoPlayerNodeDidPlayToEnd:self];
+  }
```

Comment:
Need some default action here: Seek to the first frame, or repeat automatically if the flag is true, or show the play button on the video node.